src/test/java/com/fasterxml/jackson/core/io/NumberOutputTest.java

```diff
@@ -0,0 +1,19 @@
+package com.fasterxml.jackson.core.io;
+
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class NumberOutputTest {
+    @Test
+    @Ignore
+    public void testDivBy1000() {
+        for (long i = 0; i <= Integer.MAX_VALUE; i++) {
+            int number = (int) i;
+            int expected = number / 1000;
+            int actual = NumberOutput.divBy1000(number);
```

Comment:
can you replace this test by 1 that tests a few common cases and some edge cases - you should also test the FourFifths code.

Reply:
That's why I market the tests as ignored. **They are exhaustive to be really sure, that the changes are valid not to break the code** when the change is really tiny and so important. Do you have some suggestion what values are important for this case? **What I can imagine is small values including zero, some (randomly selected?) from the middle and values near `Integer.MAX_VALUE`,** but still I have to say that so reduced tests are not enough to persuade at least me, that the code works.

EOLs I will add of course.

Reply:
an ignored test is a useless test - a test that validates thousands of values is not a test that we can add to CI

Reply:
no test is a -1 from me

Reply:
**These test just only verify the validity of the changes and I agree they should not be included in CI.** I expected, that some responsible person will use them for verification only and remove them from repository before merge.

For this task is hard to imagine, for which numbers the results differ to ones returned by original code and thus what values are important. There is many tasks like this, where is convenient to perform some exhaustive testing.
Someone with more than basic mathematical background can predict them via manual calculation, but how you know that he did not some mistake?

I can easily rework them to be included in CI, but as I said the tests if they do not test whole integer range are not worthy.

Reply:
Tests that are not included by CI are, IMO, not very useful. So I agree with @pjfanning .
But I guess I can see the point of one-off before merging.

Still, in addition, I'd probably run through 1 .. 1M, them sampling, using `Random` or something.

Reply:
Yep, random select is good also as there is not only one number, which does not fit well when the values is greater than some threshold for selected magical constants for multiplication and bitshift. In fact the values for which is expression gives invalid results are repeating with period of divisor.